### PR TITLE
Orca Event Collector

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-5077)

## Description
Adding an event collector to Orca integration.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector_description.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Orca Security
+This section explains how to fully configure the instance of Orca Security in Cortex XSOAR.
+### Integration flow
+XSOAR will query for Orca Alerts/ Alert updates every 1 minute
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector_description.md:4]: This is not right, this is not XSOAR integration - remove

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import demistomock as demisto
+import json
+import io
+from freezegun import freeze_time
+
+
+# from OrcaEventCollector import Client
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector_test.py:7]: If not needed, remove

File: Packs/Orca/ReleaseNotes/2_0_3.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### New: Orca Event Collector
+
+- Agentless, Workload-Deep, Context-Aware Security and Compliance for AWS, Azure, and GCP. (Available from Cortex XSOAR 6.8.0).
```
Comment (Shellyber) [inline Packs/Orca/ReleaseNotes/2_0_3.md:6]:
```suggestion
- Agentless, Workload-Deep, Context-Aware Security and Compliance for AWS, Azure, and GCP.
```

File: Packs/Orca/pack_metadata.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
     "name": "Orca",
     "description": "Integrate with Orca security for bidirectional incident management and fetching of asset information. \n",
     "support": "partner",
-    "currentVersion": "2.0.1",
+    "currentVersion": "2.0.3",
```
Comment (Shellyber) [inline Packs/Orca/pack_metadata.json:5]:
```suggestion
    "currentVersion": "2.1.0",
```
Change also the release notes file name

File: Packs/Orca/Integrations/OrcaEventCollector/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+## Configure Orca Event Collector on Cortex XSOAR
+
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
+2. Search for Orca Event Collector.
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | API Token | The API Key to use for connection | True |
+    | Server URL (for example: https://app.eu.orcasecurity.io/api) |  | True |
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/README.md:13]: Where is the note about the server URL? all the regions and options?

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector_description.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+   - Enter a description
+   - Mark or clear Never Expire: Never expire is marked by default. The expiration date can be configured by clearing the checkbox.
+   - Mark or clear Public: You can define public tokens that are not linked to a specific user. The token is scoped according to the user that created them but can still be used if the user is removed from the organization.
+     Select a role. 
+   - Select the accounts that you want users to have access to 
+3. Click **Create Token**.
+4. Copy the token and click Continue
+5. In the Cortex XSIAM configuration window:
+    - Paste the Api Token copied from Orca.
+    - Enter the server url.
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector_description.md:20]: Also here, lets add the note regarding the server_url

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        last_fetch: int - The time and date of the last fetch alert
+        next_page_token: str - The token to the next page.
+    Returns:
+        - list of alerts
+        - next_page_token if exist
+    """
+    response = client.get_alerts_request(max_fetch, last_fetch, next_page_token)
+    next_page_token = response.get('next_page_token')
+    alerts = response.get('data', [])
+    demisto.info(f'in get alerts {next_page_token=} , {len(alerts)=}\n {alerts=}')
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py:80]:
```suggestion
    demisto.debug(f'Get Alerts Response {next_page_token=} , {len(alerts)=}\n {alerts=}')
```

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        """
+        params = {
+            'limit': max_fetch,
+            'dsl_filter': "{\n\"filter\":\n[\n{\n\"field\": \"state.last_updated\",\n\"range\": {\n\"gt\": "
+                          "\"" + last_fetch + "\"\n}\n}\n],\n"
+                                              "\"sort\":\n[\n{\"field\":\"state.last_updated\",\n\"order\":\"asc\"\n}\n]}"
+        }
+        if next_page_token:
+            params['next_page_token'] = next_page_token
+        demisto.info(f'in get_alerts request {params=}')
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py:43]:
```suggestion
        demisto.info(f'In get_alerts request {params=}')
```

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+    return alerts, next_page_token
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py:90]: remove

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+    :rtype:
+    """
+    command = demisto.command()
+    api_token = demisto.params().get('credentials', {}).get('password')
+    server_url = demisto.params().get('server_url', 'https://app.eu.orcasecurity.io/api')
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py:95]: No need for default here, please remove

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+    proxy = demisto.params().get('proxy', False)
+
+    # How much time before the first fetch to retrieve events
+    first_fetch_time = arg_to_datetime(
+        arg=first_fetch,
+        arg_name='First fetch time',
+        required=True
+    )
+    first_fetch_time = first_fetch_time.strftime(DATE_FORMAT) if first_fetch_time else ''
+    demisto.info(f'{first_fetch_time=}')
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py:108]: Change to debug in all relevant places

File: Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+            return_results(test_module(client, last_fetch))
+        elif command in ('fetch-events', 'orca-security-get-events'):
+            alerts, next_page_token = get_alerts(client, max_fetch, last_fetch, next_page_token)
+            demisto.info(f'after get alerts {len(alerts)=} , {next_page_token=}')
+
+            if command == 'fetch-events':
+                should_push_events = True
+                current_last_run = {
+                    'next_page_token': next_page_token,
+                    'lastRun': alerts[-1].get('state', {}).get('last_updated')[:-6] if alerts else last_fetch
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/OrcaEventCollector.py:141]: explain this please

**Review by Shellyber — CHANGES_REQUESTED** [review]
Nice Work! See my notes :) 
Also, Change to demisto.debug where necessary 
What about the permissions for this API?

**Review by Shellyber — COMMENTED** [review]
nice

File: Packs/Orca/Integrations/OrcaEventCollector/Pipfile [hunk 13]
```diff
@@ -0,0 +1,18 @@
+[[source]]
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/Pipfile:1]: No need for this file and the Pipfile.lock file - you can remove both

File: Packs/Orca/Integrations/OrcaEventCollector/command_examples
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/command_examples:1]: Also remove this

**Comment by ShirleyDenkberg** [comment]
@Shellyber Doc review completed.

File: Packs/Orca/ReleaseNotes/2_1_0.md [hunk 14]
```diff
@@ -0,0 +1,10 @@
+
+#### Integrations
+
+##### New: Orca Event Collector
+
+- Agentless, Workload-Deep, Context-Aware Security and Compliance for AWS, Azure, and GCP.
```
Comment (Shellyber) [inline Packs/Orca/ReleaseNotes/2_1_0.md:6]: Fix the description according to the yml

File: Packs/Orca/Integrations/OrcaEventCollector/README.md [hunk 15]
```diff
@@ -0,0 +1,43 @@
+Agentless, Workload-Deep, Context-Aware Security and Compliance for AWS, Azure, and GCP.
```
Comment (Shellyber) [inline Packs/Orca/Integrations/OrcaEventCollector/README.md:1]: Fix the description according to yml

**Review by Shellyber — APPROVED** [review]
Nice work!

**Pull request merged**